src/NuGet.Core/NuGet.Packaging/Signing/Verification/ClientPolicyContext.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Common;
+using NuGet.Configuration;
+
+namespace NuGet.Packaging.Signing
+{
+    public class ClientPolicyContext
+    {
+        /// <summary>
+        /// Current policy the client is on.
+        /// </summary>
+        public SignatureValidationMode Policy { get; }
+
+        /// <summary>
+        /// Verification settings corresponding the current client policy.
+        /// </summary>
+        public SignedPackageVerifierSettings VerifierSettings { get; }
+
+        /// <summary>
+        /// List of signatures allowed in verification.
+        /// </summary>
+        public IReadOnlyCollection<VerificationAllowListEntry> AllowList { get; }
+
+        /// <summary>
+        /// Require AllowList to not be null or empty
+        /// </summary>
+        public bool RequireNonEmptyAllowList { get; }
```

Comment:
Do you really need this?

Doesn't SignatureValidationMode already tell you whether it can non-empty?

Reply:
It's not strictly necessary since we calculate it from the `SignatureValidationMode` ([here](https://github.com/NuGet/NuGet.Client/pull/2454/files/60b87362eaa7b49fe472dc0059e268eed2f322b0#diff-5d52f0696bf586b32763845695660a10R47)). I thought it was good for the sake of readability. Do you think we should remove this and just use the `Policy` attribute for this?

Reply:
Personally I'd prefer that.